axon-framework/queries/query-dispatchers.md

```diff
@@ -227,5 +227,132 @@ commandGateway.sendAndWait(new RedeemCmd("gc1", amount));
+
+Natively, if we want fine-grained control of the producing stream we can use Flux as return type:
+
+```java
+@QueryHandler
+public Flux<CardSummary> handle(FetchCardSummariesQuery query) {
+        ...
+    return reactiveCardRepository.findAll(); 
+}
+```
+
+When using Flux as return type, we can control backpressure, stream cancellation and implement more complex features like pagination.
+
+> **Note**
+>
+> Backpressure and stream cancellation features are only available with Axon Server 4.6 + version. Streaming queries used with pre Axon Server 4.6 version will work, but without these important features.
+>
+
+> **Transaction leaking**
+>
+> Once consumer of streaming query receives Flux to subscribe to, transaction will be considered completed successfully. That means, that any subsequent messages on the stream will not be part of transaction. 
+> This includes errors, and as transaction is already over, error will not be propagated to parent transaction to invoke any rollback method.
+> This has implication that streaming query should not be used within Unit Of Work (within message handlers or any other transactional methods) to chaining other transactional actions (like sending a command or query).
+>
+
+#### Backpressure
+Backpressure is important feature in reactive systems that allows consumer to control the flow of data, and not to be overwhelmed by the producer.
+The streaming query implements pull-based back-pressure strategy, which means that the producer will emit data when consumer is ready to receive it.
+
+Under the hood, backpressure does `Hop to Hop` signal propagation (see below) and inherits gRPC's [HTTP2-based backpressure model](https://developers.google.com/web/fundamentals/performance/http2/#flow_control).
+
+As a result, backpressure will not behave intuitively and will not propagate exact request signal from consumer to producer. 
+HTTP/2 and Netty flow control has internal buffers based on message size, Axon Framework and Axon Server prefetch messages into internal buffers based on message count. 
+Result is that the producer will send number of messages until it fills all the buffers, then backpressure will kick in.
+
+> **Hop to hop**
+>
+>Back-pressure signal is propagated per hop which is not an end to end connection,
+> which allows intermediate Axon Server to handle backpressure between two connections
+> and pre-fetch additional messages to increase overall performance.
+>
+
+#### Cancellation
+The Streaming query can be implemented as infinitive stream, so it's important to cancel it when the client is not interested in receiving any more data.
+Use operators that can cancel stream to signal producer that you are not interested in receiving more data.
+
+It's important to note that similar to backpressure, cancellation signal is per hop, meaning it's propagated over the network to Axon Server and then to producer, which can produce some latency in stream cancellation.
```

Comment:
Alright, I spot that we're again focused on Axon Server, as does the following `Error Handling` section.

A similar point should be made here: we can't assume users use Axon Server in the Axon Framework section of the Reference Guide.
Hence we need to introduce that fact.
Furthermore, some guidance on these three topics **without** Axon Server should be present too, if applicable.